setup/plugins/installFiles/l10nDownload.civi-setup.php

```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * @file
+ *
+ * Generate the civicrm.settings.php file.
+ */
+
+if (!defined('CIVI_SETUP')) {
+  exit("Installation plugins must only be loaded by the installer.\n");
+}
+
+/**
+ * Validate the $model.
+ */
+\Civi\Setup::dispatcher()
+  ->addListener('civi.setup.checkRequirements', function(\Civi\Setup\Event\CheckRequirementsEvent $e) {
+    $lang = \Civi\Setup::instance()->getModel()->lang;
+    if ($lang || $lang != 'en_US') {
+      $civicrmMoFileUrl = 'https://download.civicrm.org/civicrm-l10n-core/mo/' . $lang . '/civicrm.mo';
+      $dir = $e->getModel()->srcPath . DIRECTORY_SEPARATOR . 'l10n' . DIRECTORY_SEPARATOR . $lang . DIRECTORY_SEPARATOR . 'LC_MESSAGES';
+      \Civi\Setup::log()->info("Creating directry: " . $dir);
+      \CRM_Utils_File::createDir($dir, FALSE);
```

Comment:
If running via the web page, this might fail on sites with tighter security where the webserver user doesn't have write access to the code files tree. It should maybe log a warning and skip the remaining steps.

Reply:
I have added error handling